content/posts/2021/01/2021-11-04-Getting-Started-With-React/index.md

```diff
@@ -0,0 +1,334 @@
+[Development Environment]()
+[Creating a React App]()
+[Exploring Create React App]()
+[Our First Component]()
+[JSX]()
+[How a Component Gets Rendered]()
+[Let's Display Data Fetched From an API]()
+
+
+## Prerequisites
+
+
+You don't need to have any prior knowldge of React before reading this article, you however need to be familiar with a few things in order to get the most out of this guide.
+
+#### Basic JavaScript
+
+
+React is a JS library so it makes sense to know JavaSript before learning React right? Worry less, you __don't__ need to know Javscript extensively to learn React Js, just the basics.Here's a list of some of the concepts you should be familiar with before learning React.
+- Variables, functions and Data Types
+- Asychronous Programming
+- Arrays and Objects
+- ES6 features and syntax
+- DOM Manupulation
+
+#### Basic HTML
+
+In React we use JSX, which we will dive into later. But for now have a good grasp of HTML preferably HTML5.
+
+- How to structure HTML elements.
+- HTML attributes such as class, id, onclick, href etc
+
+## Development Environment
+
+The first thing we do before interacting with any programming language is to set up it's developemt environment, React JS is no different.
+If you have **Node JS** and **Visual Studio Code** (or your preffered IDE) set up you can skip this section.
+
+#### Node JS
+
+Do I really need node to set up a React app? The answer is no. If you are a web developer that can handle HTML/CSS and JavaSCript you can try React without Node JS or any other complext tool to manage it. That's your assignment.
+
+In our case we need Node js. To download it visit [<span style="color:blue">**this link**</span>]( https://nodejs.org/en/download/ ) and choose one that fit's your machine.
+
+When the installation is complete, open your terminal and type 
+ > node -v 
+
+This should show you the version of node that you have installed.
+The output should look something like:
+ > v16.7.0
+
+Hurrraaaaaayy!!! Congratulations! You have completed the first step to getting started with React.
+
+
+#### Visual Studio Code
+
+VSC is a popular Open Source IDE for frontend development. There's a bunch of IDEs that you can try depending on your preference. For now we will run with Visual Studio Code.
+
+Click [<span style="color:blue">**this link**</span>]( https://code.visualstudio.com/download ) to download the version that suits your platform.
+That's enough development set up for now.
+
+
+## Creating a React App
+
+The next step is now creating a React app. Thank God for the folks at Facebook who made it easy for us to do that.
+All we to do is run this command in our terminal.
+
+ > npx create-react-app my-app
+
+This craetes a React project called my-app.
+![Output on the terminal](./images/vscode.png)
+
+Realize we've been directed on how to start our app
+ > cd my-app
+ > yarn start
+
+
+ Run this commands on your terminal, this will start a development server and open a web browser for you.
```

Comment:
- Image missing due to wrong path reference. Image is inside `./react-app.png` since it's in the same directory level as the markdown file.

- Also typo `this command` not `this commands`

![Screenshot 2021-11-28 at 23 13 03](https://user-images.githubusercontent.com/9129843/143784312-ed71a18c-6e3e-4e46-9316-0be5ed7d401d.png)